src/util/entropyCreateStateFromJsons.ts

```diff
@@ -280,17 +314,30 @@ function cdsFromAnnotation(cdsName: string, annotation: JsonAnnotation, rangeGen
  * Calculates the (open reading) frame the provided segment is in.
  * For +ve strand this is calculated 5'->3', for -ve strand it's 3'->5'.
  * The frame is calculated once the CDS is back in phase.
- * start: 1 based, rangeGenome[0] of the segment
- * end: 1 based, rangeGenome[1] of the segment
- * start < end always
- * genomeLength: 1 based
- * positiveStrand: boolean
- * Returns a number in the set {0, 1, 2}
  */
-function _frame(start:number, end:number, phase: number, genomeLength:number, positiveStrand:boolean) {
-  return positiveStrand ?
+function _frame(
+  /** 1 based, rangeGenome[0] of the segment */
+  start: number,
+
+  /**
+   * 1 based, rangeGenome[1] of the segment.
+   * start < end always
+   */
+  end: number,
+
+  phase: Phase,
+
+  /** 1 based */
+  genomeLength: number,
+  positiveStrand: boolean,
+): Frame {
+  /* TypeScript cannot infer the exact range of values from a modulo operation,
+   * so it is manually provided.
+   */
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+  return (positiveStrand ?
     (start+phase-1)%3 :
-    Math.abs((end-phase-genomeLength)%3);
+    Math.abs((end-phase-genomeLength)%3)) as 0 | 1 | 2;
```

Comment:
I think this should work and it makes it more explicit what's going on...

```suggestion
    Math.abs((end-phase-genomeLength)%3)) as Frame;
```

Reply:
My thinking is that `0 | 1 | 2` is logically the outcome of `%3` – if TypeScript were smart enough, it would know that without needing the cast. The return type of `Frame` in the function signature will ensure that `0 | 1 | 2` can be properly casted to `Frame`.

Does that make sense or do you still prefer `as Frame` here?

Reply:
Personally, I would write it `as Frame`; the comment makes it clear why the cast is there (Typescript can't infer the mod result, fine), and I don't have to go and look at what the type definition of `Frame` actually is. 🤷 

Non-blocking/weak opinion.

Reply:
Ok, I'll keep as-is unless there is another vote for `as Frame`.